rd_ui/app/scripts/directives/directives.js

```diff
@@ -58,7 +58,7 @@
         }
 
         $scope.$on('$locationChangeStart', function (event, next, current) {
-          if (next.split("#")[0] == current.split("#")[0]) {
+          if (next.split("?")[0] == current.split("?")[0]) {
```

Comment:
We need the previous check (with the `#`) too, for cases when the user changes the current visualization and there are no parameters present (and hence no query string parameters).